lib/cfg-tree.c

```diff
@@ -476,8 +478,8 @@ cfg_tree_compile_single(CfgTree *self, LogExprNode *node,
           goto error;
         }
       pipe->flags |= PIF_INLINED;
+      g_ptr_array_add(self->initialized_pipes, pipe);
```

Comment:
We talked it IRL, I let the `g_ptr_array_add` in the original place, and call only `log_pipe_ref(pipe)` on line 468.